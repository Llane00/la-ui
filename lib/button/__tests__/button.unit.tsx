import renderer from 'react-test-renderer'
import React from 'react'
import Button from '../button'
import { scopedClassMaker } from '../../helpers/classes'

describe('button', () => {
    it('button组件测试', ()=>{
      const json = renderer.create(<Button btnType='default'>测试</Button>).toJSON()
      expect(json).toMatchSnapshot()
    })
})

describe('scopedClassMaker', () => {
  it('', ()=>{
    const sc = scopedClassMaker('la-ui-button');
    expect(sc('')).toEqual('la-ui-button');
    expect(sc('x')).toEqual('la-ui-button-x');
    expect(sc({'': true})).toEqual('la-ui-button');
    expect(sc({'': true, 'default': false})).toEqual('la-ui-button');
    expect(sc({'': true, 'default': false}, {extra: 'red'})).toEqual('la-ui-button red');
  })
})
