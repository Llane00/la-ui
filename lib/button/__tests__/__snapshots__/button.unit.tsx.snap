// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button button组件测试 1`] = `
<button
  className="la-ui-button"
>
  测试
</button>
`;
